src/daft-dsl/src/expr/mod.rs

```diff
@@ -664,6 +676,73 @@ impl From<&AggExpr> for ExprRef {
     }
 }
 
+impl WindowExpr {
+    pub fn name(&self) -> &str {
+        match self {
+            Self::Agg(agg_expr) => agg_expr.name(),
+            Self::Rank() => "rank()",
+        }
+    }
+
+    pub fn semantic_id(&self, schema: &Schema) -> FieldID {
+        match self {
+            Self::Agg(agg_expr) => agg_expr.semantic_id(schema),
+            Self::Rank() => FieldID::new("rank()"),
+        }
+    }
+
+    pub fn children(&self) -> Vec<ExprRef> {
+        match self {
+            Self::Agg(agg_expr) => agg_expr.children(),
+            Self::Rank() => vec![],
+        }
+    }
+
+    pub fn with_new_children(&self, children: Vec<ExprRef>) -> Self {
+        match self {
+            Self::Agg(agg_expr) => Self::Agg(agg_expr.with_new_children(children)),
+            Self::Rank() => Self::Rank(),
+        }
+    }
+
+    pub fn to_field(&self, schema: &Schema) -> DaftResult<Field> {
+        match self {
+            Self::Agg(agg_expr) => agg_expr.to_field(schema),
+            Self::Rank() => Ok(Field::new("rank()", DataType::UInt64)),
+        }
+    }
+}
+
+impl From<&WindowExpr> for ExprRef {
+    fn from(window_expr: &WindowExpr) -> Self {
+        match window_expr {
+            WindowExpr::Agg(agg_expr) => Self::new(Expr::Agg(agg_expr.clone())),
+            _ => Self::new(Expr::Window(window_expr.clone())),
+        }
+    }
+}
```

Comment:
The behavior here may be confusing since it unnests aggs. I would suggest having the conversion simply being Arc::new(Expr::WindowFunction(window_expr)) for all cases